src/nmodl/main.cpp

```diff
@@ -329,6 +330,17 @@ int main(int argc, const char* argv[]) {
             UnitsVisitor(units_dir).visit_program(ast.get());
         }
 
+        /// GLOBAL to RANGE rename visitor
+        {
+            // make sure to run perf visitor because code generator
+            // looks for read/write counts const/non-const declaration
+            PerfVisitor().visit_program(ast.get());
```

Comment:
I see that now the `PerfVisitor` is being run 3 to 4 times.. do we really have to do this?

Reply:
Well, it was needed before the `CodegenCompatibilityVisitor` to check the symtab write count for `GLOBAL` variables and decide if there must be generated an error about turning them to `RANGE`. If we want to delete this case from the `CodegenCompatibilityVisitor` we can remove this `PerfVisitor` pass. 
Then it is needed for sure before the `GlobalToRangeVisitor`. 
I am not sure if the last call before the code generation can be removed due to the `SymtabVisitor(update_symtab).visit_program(ast.get());` calls that are happening before it. Based on the tests I've made, the last pass of the `PerfVisitor` can be removed and still get the same code generated.

Reply:
@pramodk can you way in on that? Do we need all those calls to that visitor?

Reply:
> I see that now the PerfVisitor is being run 3 to 4 times.. do we really have to do this?

Let's address first question : when + why PerfVisitor needs to be run?
-> When perfvisitor runs, it update all operations count associated with symbols. If particular pass is dependent on such ops-count, perfvisitor must be run. In the current scenario, by default, PerfVisitor is run twice : codegen compatibility and code generation. 

Why we need to run it multiple times?
-> Passes like Inlining and localiser change the AST and hence symbol table needs to be updated with associated ops-count. And hence they need to be re-run.

Actually, PerfVisitor should be run by the associated pass but lets do that re-factorign in separate ticket.

Reply:
@pramodk Okay. So the `PerfVisitor` pass before the codegen should stay. 
Another question I have is if the Global to Range error in the `CodegenCompatibilityVisitor` should stay, since you told me that there will be some inconsistency between `Neuron` and `CoreNeuron` data with this transformation.

Reply:
> Another question I have is if the Global to Range error in the CodegenCompatibilityVisitor should stay, since you told me that there will be some inconsistency between Neuron and CoreNeuron data with this transformation.

yes, that's true.